### ISSUE TITLE
 The CUDA linking API did not work.

### ISSUE THREAD

**Issue by GravityPHY** [google/jax#16063]
### Description

```
>>> import jax
>>> import jax.numpy as jnp
>>> x=jnp.arange(10)
```
Giving me the error

E external/xla/xla/service/gpu/gpu_compiler.cc:1592] The CUDA linking API did not work. Please use XLA_FLAGS=--xla_gpu_force_compilation_parallelism=1 to bypass it, but expect to get longer compilation time due to the lack of multi-threading. Original error: INTERNAL: nvlink exited with non-zero error code 256, output: nvlink fatal   : Input file '/tmp/tempfile-giovanni-8ba750b0-2357013-5fbfed708a26f.cubin' newer than toolkit (118 vs 117)

### What jax/jaxlib version are you using?

jax 0.4.8, jaxlib 0.4.7

### Which accelerator(s) are you using?

GPU

### Additional system info

Python 3.8.0

### NVIDIA GPU info

```
Thu May 18 17:57:40 2023       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 515.65.01    Driver Version: 515.65.01    CUDA Version: 11.7     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA RTX A6000    Off  | 00000000:81:00.0 Off |                  Off |
| 39%   67C    P2   105W / 300W |  10575MiB / 49140MiB |    100%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA RTX A6000    Off  | 00000000:82:00.0 Off |                  Off |
| 53%   77C    P2   178W / 300W |  22864MiB / 49140MiB |     22%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  NVIDIA RTX A6000    Off  | 00000000:C1:00.0 Off |                  Off |
| 36%   63C    P8    39W / 300W |      8MiB / 49140MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  NVIDIA RTX A6000    Off  | 00000000:C2:00.0 Off |                  Off |
| 33%   61C    P8    40W / 300W |    108MiB / 49140MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      2282      G   /usr/lib/xorg/Xorg                  4MiB |
|    1   N/A  N/A      2282      G   /usr/lib/xorg/Xorg                 64MiB |
|    1   N/A  N/A      2592      G   /usr/bin/gnome-shell                7MiB |
|    1   N/A  N/A   2341935      C   python                          22741MiB |
|    2   N/A  N/A      2282      G   /usr/lib/xorg/Xorg                  4MiB |
|    3   N/A  N/A      2282      G   /usr/lib/xorg/Xorg                  4MiB |
+-----------------------------------------------------------------------------+
```

**Comment by hawkinsp** [comment]
How did you install JAX and CUDA?

It appears JAX is finding too old a version of `nvlink`, or mismatched versions of `nvlink` and `ptxas`. JAX requires CUDA 11.8.

**Comment by GravityPHY** [comment]
I installed JAX and JAXlib by 
`pip install "jax[cuda11_cudnn82]" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html`

**Comment by GravityPHY** [comment]
The CUDA version is 11.7. Is it upgrade to CUDA 11.8 the only solution?

**Comment by GravityPHY** [comment]
> The CUDA version is 11.7. Is it upgrade to CUDA 11.8 the only solution?

In my case, I install CUDA 11.8 by `conda install -c "nvidia/label/cuda-11.8.0" cuda-toolkit` and this issue disappear.

**Comment by hawkinsp** [comment]
Yes, JAX no longer ships wheels with support for CUDA versions older than 11.8. Hope that helps!
